### DISCUSSION TITLE
Got this error "ModuleNotFoundError: No module named 'diffusers'" what should I do?

### DISCUSSION THREAD

**Discussion by songib** [lshqqytiger/stable-diffusion-webui-directml#190] [Q&A] [answered]
`Already up to date.
The filename, directory name, or volume label syntax is incorrect.
venv "C:\Art Software\Stable Diffusion WebUI (AI ART)\stable-diffusion-webui-directml\venv\Scripts\Python.exe"
fatal: No names found, cannot describe anything.
Python 3.10.6 (tags/v3.10.6:9c7b4bd, Aug  1 2022, 21:53:49) [MSC v.1932 64 bit (AMD64)]
Version: ## 1.4.1
Commit hash: 6789a1f3d1cf039492ac008b107ed01c7aa1ebe6
Installing requirements

Checking roop requirements
Install insightface==0.7.3
Installing sd-webui-roop requirement: insightface==0.7.3
Install onnx==1.14.0
Installing sd-webui-roop requirement: onnx==1.14.0
Install onnxruntime==1.15.0
Installing sd-webui-roop requirement: onnxruntime==1.15.0
Install opencv-python==4.7.0.72
Installing sd-webui-roop requirement: opencv-python==4.7.0.72

Launching Web UI with arguments: --no-half --no-half-vae --precision full --always-batch-cond-uncond --medvram --api --autolaunch --disable-nan-check --theme dark --use-cpu CLIP BLIP interrogate
No module 'xformers'. Proceeding without it.
╭─────────────────────────────── Traceback (most recent call last) ────────────────────────────────╮
│ C:\Art Software\Stable Diffusion WebUI (AI ART)\stable-diffusion-webui-directml\launch.py:38 in  │
│ <module>                                                                                         │
│                                                                                                  │
│   35                                                                                             │
│   36                                                                                             │
│   37 if __name__ == "__main__":                                                                  │
│ ❱ 38 │   main()                                                                                  │
│   39                                                                                             │
│                                                                                                  │
│ C:\Art Software\Stable Diffusion WebUI (AI ART)\stable-diffusion-webui-directml\launch.py:34 in  │
│ main                                                                                             │
│                                                                                                  │
│   31 │   if args.test_server:                                                                    │
│   32 │   │   configure_for_tests()                                                               │
│   33 │                                                                                           │
│ ❱ 34 │   start()                                                                                 │
│   35                                                                                             │
│   36                                                                                             │
│   37 if __name__ == "__main__":                                                                  │
│                                                                                                  │
│ C:\Art Software\Stable Diffusion WebUI (AI                                                       │
│ ART)\stable-diffusion-webui-directml\modules\launch_utils.py:361 in start                        │
│                                                                                                  │
│   358                                                                                            │
│   359 def start():                                                                               │
│   360 │   print(f"Launching {'API server' if '--nowebui' in sys.argv else 'Web UI'} with argum   │
│ ❱ 361 │   import webui                                                                           │
│   362 │   if '--nowebui' in sys.argv:                                                            │
│   363 │   │   webui.api_only()                                                                   │
│   364 │   else:                                                                                  │
│                                                                                                  │
│ C:\Art Software\Stable Diffusion WebUI (AI ART)\stable-diffusion-webui-directml\webui.py:42 in   │
│ <module>                                                                                         │
│                                                                                                  │
│    39 startup_timer.record("import ldm")                                                         │
│    40                                                                                            │
│    41 from modules import extra_networks                                                         │
│ ❱  42 from modules.call_queue import wrap_gradio_gpu_call, wrap_queued_call, queue_lock  # noq   │
│    43                                                                                            │
│    44 # Truncate version number of nightly/local build of PyTorch to not cause exceptions with   │
│    45 if ".dev" in torch.__version__ or "+git" in torch.__version__:                             │
│                                                                                                  │
│ C:\Art Software\Stable Diffusion WebUI (AI                                                       │
│ ART)\stable-diffusion-webui-directml\modules\call_queue.py:5 in <module>                         │
│                                                                                                  │
│     2 import threading                                                                           │
│     3 import time                                                                                │
│     4                                                                                            │
│ ❱   5 from modules import shared, progress, errors                                               │
│     6                                                                                            │
│     7 queue_lock = threading.Lock()                                                              │
│     8                                                                                            │
│                                                                                                  │
│ C:\Art Software\Stable Diffusion WebUI (AI                                                       │
│ ART)\stable-diffusion-webui-directml\modules\shared.py:86 in <module>                            │
│                                                                                                  │
│    83                                                                                            │
│    84                                                                                            │
│    85 if cmd_opts.backend == 'directml':                                                         │
│ ❱  86 │   directml_hijack_init()                                                                 │
│    87 if device.type == 'privateuseone':                                                         │
│    88 │   directml_init()                                                                        │
│    89                                                                                            │
│                                                                                                  │
│ C:\Art Software\Stable Diffusion WebUI (AI                                                       │
│ ART)\stable-diffusion-webui-directml\modules\dml\__init__.py:21 in directml_hijack_init          │
│                                                                                                  │
│   18 │   torch.cuda.reset_peak_memory_stats = torch.dml.reset_peak_memory_stats                  │
│   19                                                                                             │
│   20 def directml_hijack_init():                                                                 │
│ ❱ 21 │   import modules.dml.hijack                                                               │
│   22                                                                                             │
│                                                                                                  │
│ C:\Art Software\Stable Diffusion WebUI (AI                                                       │
│ ART)\stable-diffusion-webui-directml\modules\dml\hijack\__init__.py:6 in <module>                │
│                                                                                                  │
│   3 import modules.dml.hijack.torch                                                              │
│   4 import modules.dml.hijack.realesrgan_model                                                   │
│   5 import modules.dml.hijack.plms                                                               │
│ ❱ 6 import modules.dml.hijack.diffusers                                                          │
│   7 import modules.dml.hijack.transformers                                                       │
│   8                                                                                              │
│                                                                                                  │
│ C:\Art Software\Stable Diffusion WebUI (AI                                                       │
│ ART)\stable-diffusion-webui-directml\modules\dml\hijack\diffusers.py:2 in <module>               │
│                                                                                                  │
│    1 from torch import FloatTensor                                                               │
│ ❱  2 import diffusers                                                                            │
│    3                                                                                             │
│    4 def _get_prev_sample(self, sample: FloatTensor, timestep, prev_timestep, model_output):     │
│    5 │   # See formula (9) of PNDM paper https://arxiv.org/pdf/2202.09778.pdf                    │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
ModuleNotFoundError: No module named 'diffusers'
Press any key to continue . . .`

**Comment by Synfinite** [comment]
Looks like @lshqqytiger merged some code that's causing the error, I'm able to work around it by commenting out line 2 and line 48 in diffusers.py for the time being, until a fix is implemented.
![image](https://github.com/lshqqytiger/stable-diffusion-webui-directml/assets/37858559/556295f6-0166-492e-96e6-1c101d5b9ef8)

**Reply by songib** [reply]
Ahh, I will wait for the update then. ty

**Comment by lshqqytiger** [comment, marked as answer]
Should be fixed in 02b316ee3ba135f124060795cb0182f6d70033d0.

**Reply by songib** [reply]
TY. Have a nice day.

**Reply by songib** [reply]
@lshqqytiger Still got the error.

Already up to date.
The filename, directory name, or volume label syntax is incorrect.
venv "C:\Art Software\Stable Diffusion WebUI (AI ART)\stable-diffusion-webui-directml\venv\Scripts\Python.exe"
fatal: No names found, cannot describe anything.
Python 3.10.6 (tags/v3.10.6:9c7b4bd, Aug  1 2022, 21:53:49) [MSC v.1932 64 bit (AMD64)]
Version: ## 1.4.1
Commit hash: 99e93f174f783f4848bddeff7cc0d82dacc20719
Installing requirements




Checking roop requirements
Install insightface==0.7.3
Installing sd-webui-roop requirement: insightface==0.7.3
Install onnx==1.14.0
Installing sd-webui-roop requirement: onnx==1.14.0
Install onnxruntime==1.15.0
Installing sd-webui-roop requirement: onnxruntime==1.15.0
Install opencv-python==4.7.0.72
Installing sd-webui-roop requirement: opencv-python==4.7.0.72

Launching Web UI with arguments: --no-half --no-half-vae --precision full --always-batch-cond-uncond --medvram --api --autolaunch --disable-nan-check --theme dark --use-cpu CLIP BLIP interrogate
No module 'xformers'. Proceeding without it.
╭─────────────────────────────── Traceback (most recent call last) ────────────────────────────────╮
│ C:\Art Software\Stable Diffusion WebUI (AI ART)\stable-diffusion-webui-directml\launch.py:38 in  │
│ <module>                                                                                         │
│                                                                                                  │
│   35                                                                                             │
│   36                                                                                             │
│   37 if __name__ == "__main__":                                                                  │
│ ❱ 38 │   main()                                                                                  │
│   39                                                                                             │
│                                                                                                  │
│ C:\Art Software\Stable Diffusion WebUI (AI ART)\stable-diffusion-webui-directml\launch.py:34 in  │
│ main                                                                                             │
│                                                                                                  │
│   31 │   if args.test_server:                                                                    │
│   32 │   │   configure_for_tests()                                                               │
│   33 │                                                                                           │
│ ❱ 34 │   start()                                                                                 │
│   35                                                                                             │
│   36                                                                                             │
│   37 if __name__ == "__main__":                                                                  │
│                                                                                                  │
│ C:\Art Software\Stable Diffusion WebUI (AI                                                       │
│ ART)\stable-diffusion-webui-directml\modules\launch_utils.py:390 in start                        │
│                                                                                                  │
│   387                                                                                            │
│   388 def start():                                                                               │
│   389 │   print(f"Launching {'API server' if '--nowebui' in sys.argv else 'Web UI'} with argum   │
│ ❱ 390 │   import webui                                                                           │
│   391 │   if '--nowebui' in sys.argv:                                                            │
│   392 │   │   webui.api_only()                                                                   │
│   393 │   else:                                                                                  │
│                                                                                                  │
│ C:\Art Software\Stable Diffusion WebUI (AI ART)\stable-diffusion-webui-directml\webui.py:42 in   │
│ <module>                                                                                         │
│                                                                                                  │
│    39 startup_timer.record("import ldm")                                                         │
│    40                                                                                            │
│    41 from modules import extra_networks                                                         │
│ ❱  42 from modules.call_queue import wrap_gradio_gpu_call, wrap_queued_call, queue_lock  # noq   │
│    43                                                                                            │
│    44 # Truncate version number of nightly/local build of PyTorch to not cause exceptions with   │
│    45 if ".dev" in torch.__version__ or "+git" in torch.__version__:                             │
│                                                                                                  │
│ C:\Art Software\Stable Diffusion WebUI (AI                                                       │
│ ART)\stable-diffusion-webui-directml\modules\call_queue.py:5 in <module>                         │
│                                                                                                  │
│     2 import threading                                                                           │
│     3 import time                                                                                │
│     4                                                                                            │
│ ❱   5 from modules import shared, progress, errors                                               │
│     6                                                                                            │
│     7 queue_lock = threading.Lock()                                                              │
│     8                                                                                            │
│                                                                                                  │
│ C:\Art Software\Stable Diffusion WebUI (AI                                                       │
│ ART)\stable-diffusion-webui-directml\modules\shared.py:88 in <module>                            │
│                                                                                                  │
│    85                                                                                            │
│    86                                                                                            │
│    87 if cmd_opts.backend == 'directml':                                                         │
│ ❱  88 │   directml_hijack_init()                                                                 │
│    89 if device.type == 'privateuseone':                                                         │
│    90 │   directml_init()                                                                        │
│    91                                                                                            │
│                                                                                                  │
│ C:\Art Software\Stable Diffusion WebUI (AI                                                       │
│ ART)\stable-diffusion-webui-directml\modules\dml\__init__.py:21 in directml_hijack_init          │
│                                                                                                  │
│   18 │   torch.cuda.reset_peak_memory_stats = torch.dml.reset_peak_memory_stats                  │
│   19                                                                                             │
│   20 def directml_hijack_init():                                                                 │
│ ❱ 21 │   import modules.dml.hijack                                                               │
│   22                                                                                             │
│                                                                                                  │
│ C:\Art Software\Stable Diffusion WebUI (AI                                                       │
│ ART)\stable-diffusion-webui-directml\modules\dml\hijack\__init__.py:6 in <module>                │
│                                                                                                  │
│   3 import modules.dml.hijack.torch                                                              │
│   4 import modules.dml.hijack.realesrgan_model                                                   │
│   5 import modules.dml.hijack.plms                                                               │
│ ❱ 6 import modules.dml.hijack.diffusers                                                          │
│   7 import modules.dml.hijack.transformers                                                       │
│   8                                                                                              │
│                                                                                                  │
│ C:\Art Software\Stable Diffusion WebUI (AI                                                       │
│ ART)\stable-diffusion-webui-directml\modules\dml\hijack\diffusers.py:2 in <module>               │
│                                                                                                  │
│    1 from torch import FloatTensor                                                               │
│ ❱  2 import diffusers                                                                            │
│    3                                                                                             │
│    4 def _get_prev_sample(self, sample: FloatTensor, timestep, prev_timestep, model_output):     │
│    5 │   # See formula (9) of PNDM paper https://arxiv.org/pdf/2202.09778.pdf                    │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
ModuleNotFoundError: No module named 'diffusers'
Press any key to continue . . .

**Reply by lshqqytiger** [reply]
Try removing `venv` folder.
Or install diffusers manually:
```
(venv) $ pip install diffusers
```

**Reply by songib** [reply]
ok

**Reply by songib** [reply]
@lshqqytiger removing `venv` didn't work I did the manual install instead.

**Reply by lshqqytiger** [reply]
Strange. I already added diffusers to `requirements.txt`..

**Reply by songib** [reply]
@lshqqytiger I did the manual install and it works now. but still got the [problem with out-of-memory](https://github.com/lshqqytiger/stable-diffusion-webui-directml/discussions/189) stuff when using ADetailer and need a full restart webUI.

**Comment by dieselino0** [comment]
I am also getting this error, I tried deleting venv folder and also install diffusers manually but it still doesn't work

**EDIT**

Latest update has fixed the issue - https://github.com/lshqqytiger/stable-diffusion-webui-directml/commit/aa02899d39c3f0261db4b4b70b9ce262e61fe440

**Reply by songib** [reply]
up to date and in main brach?

**Reply by dieselino0** [reply]
Yup all up to date and using the main branch but I still have the issue
What fixed it, for now, was this - https://github.com/lshqqytiger/stable-diffusion-webui-directml/discussions/190#discussioncomment-6465331

**EDIT**

Looks like it was just updated and the newest update has fixed it

https://github.com/lshqqytiger/stable-diffusion-webui-directml/commit/aa02899d39c3f0261db4b4b70b9ce262e61fe440
